var/spack/repos/spack_repo/builtin/packages/py_pathspec/package.py

```diff
@@ -16,6 +16,9 @@ class PyPathspec(PythonPackage):
 
     license("MPL-2.0")
 
+    version("0.12.1", sha256="a482d51503a1ab33b1c67a6c3813a26953dbdc71c31dacaef9a838c4e29f5712")
```

Comment:
Confirmed the new version directives' sha256s but the minimum version of python for `@0.12:` needs to be bumped up per their `pyproject.toml` files:

```
requires-python = ">=3.8"
```

So the dependency on line 27 can be capped (`@0.10:0.11`).

The version requirement for `py-setuptools` appears in the `setup.cfg` files of the newly listed versions, each of which indicate `py-setuptools@40.8:` is still required:

```
[options]
..
setup_requires = setuptools>=40.8.0
```

despite not being specified for `@11:` (see lines 35-37).

Reply:
I completely forgot to check the transitive dependencies, so thanks for looking at this.

I think I made all the changes you requested in 79af649be6fe64b7abe6b8277d3fd7b189a26653, but let me know if I missed anything.